comp/core/workloadmeta/collectors/internal/kubeapiserver/kubeapiserver.go

```diff
@@ -187,21 +189,15 @@ func (c *collector) GetTargetCatalog() workloadmeta.AgentType {
 	return c.catalog
 }
 
-func startReadiness(ctx context.Context, stores []*reflectorStore) {
-	log.Infof("Starting readiness waiting for %d k8s reflectors to sync", len(stores))
+func startStartup(ctx context.Context, stores []*reflectorStore) {
```

Comment:
nit: I think `startStartup` is a bit strange. Can we call it `startStartupHealthCheck` or similar?

Reply:
Yeah I was a bit unsure about the name too, `runStartupCheck` it is 👍 

```
diff --git a/comp/core/workloadmeta/collectors/internal/kubeapiserver/kubeapiserver.go b/comp/core/workloadmeta/collectors/internal/kubeapiserver/kubeapiserver.go
index ada76ec07b..68c80cbfc7 100644
--- a/comp/core/workloadmeta/collectors/internal/kubeapiserver/kubeapiserver.go
+++ b/comp/core/workloadmeta/collectors/internal/kubeapiserver/kubeapiserver.go
@@ -177,7 +177,7 @@ func (c *collector) Start(ctx context.Context, wlmetaStore workloadmeta.Componen
 		go reflector.Run(ctx.Done())
 	}
 
-	go startStartup(ctx, objectStores)
+	go runStartupCheck(ctx, objectStores)
 
 	return nil
 }
@@ -194,7 +194,7 @@ func (c *collector) GetTargetCatalog() workloadmeta.AgentType {
 	return c.catalog
 }
 
-func startStartup(ctx context.Context, stores []*reflectorStore) {
+func runStartupCheck(ctx context.Context, stores []*reflectorStore) {
 	log.Infof("Starting startup health check waiting for %d k8s reflectors to sync", len(stores))
 
 	// There is no way to ensure liveness correctly as it would need to be plugged inside the

```